git/index.go

```diff
@@ -187,7 +188,7 @@ const (
 //
 // As a special case, if something is added as Stage0, then Stage1-3 entries
 // will be removed.
-func (g *Index) AddStage(c *Client, path IndexPath, mode EntryMode, s Sha1, stage Stage, size uint32, mtime int64, createEntry bool) error {
+func (g *Index) AddStage(c *Client, path IndexPath, mode EntryMode, s Sha1, stage Stage, size uint32, mtime int64, createEntry bool, replaceEntry bool) error {
```

Comment:
The code path that uses the new param looks pretty self-contained, and only has one caller that sets it, so can we remove the bool param and have a ReplaceStage function instead?

Alternatively, if AddStage and ReplaceStage do share a lot of code, maybe AddStage should take an UpdateIndexOptions to keep it more readable at the call site.

Reply:
The arguments list is getting pretty big. Since there's a lot of shared code so far I was thinking of passing in the opts.

Reply:
You can probably replace createEntry with opts.Add if you do that, too. (I'm more concerned with bool params being hard to maintain than the number of params.)

I'm also considering replacing the size/mtime param with some kind of new StatInfo struct in #54, but I'm holding off on that for now because of conflicts with this.

Reply:
@driusan if this is the only thing holding you back from merging can you go ahead and do that? I would like to work on this a little more and validate some assumptions and include your final changes to eliminate some variables. There's a few branches that I'm working off and it's starting to get confusing. :)

Reply:
I just merged it without the change and left the FIXME note. It can be cleaned up later, for now it's fixing both diff-files remaining tests. That leaves this and rebasing the other PR to have t0000-basic.sh running.